### ISSUE TITLE
Noticeable performance drop

### ISSUE THREAD

**Issue by evshiron** [zerotier/libzt#41]
### ZeroTier Library vs ZeroTier One

We are using the `ipv4server_udp` demo included in `libzt`, and the ZeroTier One installed by `curl | bash`. Both of them run inside an Ubuntu 18.04.1 VM, then we ping from the host and notice the performance difference.

The table attached is transformed from `mtr --report`:

ZeroTier Library:

| Loss% | Snt | Last | Avg | Best | Wrst | StDev |
| --- | --- | --- | --- | --- | --- | --- |
| 0.0% | 10 | 49.3 | 33.2 | 7.6 | 50.5 | 15.1 |

ZeroTier One:

| Loss% | Snt | Last | Avg | Best | Wrst | StDev |
| --- | --- | --- | --- | --- | --- | --- |
| 0.0% | 10 | 1.1 | 0.7 | 0.4 | 1.1 | 0.2 |

**Comment by joseph-henry** [comment]
Good catch. I should document this, but try adjusting `LWIP_GUARDED_BUF_CHECK_INTERVAL` in `include/libztDefs.h` to a value more suitable for your purpose. It's a trade-off between responsiveness and CPU utilization. A more reasonable value than the default might be `5 ms`. 

Further optimizations can be made with regards to maximum numbers of connections, memory allocation, checksum functions, etc. But much of that requires one to be familiar with the `lwIP` architecture so feel free to point out any other performance bottlenecks you experience and I'll see what I can do.
